src/Psalm/Type/Atomic/TKeyedArray.php

```diff
@@ -120,7 +125,7 @@ function ($name, Union $type): string {
         }
 
         /** @psalm-suppress MixedOperand */
-        return static::KEY . '{' .
+        return ($this->non_empty ? 'non-empty-array' : static::KEY) . '{' .
```

Comment:
Should I pull this out into a private method? I was thinking about updating `getKey()` but that won't work because it has to be `'array'` since it's used for `Union::$types`. Maybe `getType()`?

Reply:
This is fine here, I think.